connector/src/main/java/it/eng/connector/service/UserService.java

```diff
@@ -0,0 +1,121 @@
+import lombok.extern.slf4j.Slf4j;
+
+@Service
+@Slf4j
+public class UserService {
+
+	private final UserRepository userRepository;
+	private final PasswordEncoder encoder;
+	private final PasswordCheckValidator passwordValidator;
+
+	public UserService(UserRepository userRepository, PasswordEncoder encoder, PasswordCheckValidator passwordValidator) {
+		super();
+		this.userRepository = userRepository;
+		this.encoder = encoder;
+		this.passwordValidator = passwordValidator;
+	}
+	
+	public Collection<JsonNode> findUsers(String email) throws ResourceNotFoundException {
+		if(StringUtils.isNotBlank(email)) {
+			User user = userRepository.findByEmail(email).orElseThrow(ResourceNotFoundException::new);
+			return Collections.singletonList(Serializer.serializePlainJsonNode(user));
+//					.ifPresentOrElse(u -> { 
+//						Serializer.serializePlainJsonNode(u);
+//					}, () -> { throw new IllegalStateException(); });
+//					.stream()
+//					.map(u -> Serializer.serializePlainJsonNode(u))
+//					.collect(Collectors.toList());
+		} 
+		return userRepository.findAll()
+				.stream()
+				.map(u -> Serializer.serializePlainJsonNode(u))
+				.collect(Collectors.toList());
+	}
+
+	public JsonNode createUser(UserDTO userDTO) {
+		// check if user exists
+		userRepository.findByEmail(userDTO.getEmail())
+			.ifPresent(u -> {
+				throw new BadRequestException("User with email already exists");
+				});
+		PasswordValidationResult validationResult = passwordValidator.isValid(userDTO.getPassword());
+		if(validationResult.isValid()) {
+			User user = new User(createNewPid(), userDTO.getFirstName(), userDTO.getLastName(), userDTO.getEmail(), 
+					encoder.encode(userDTO.getPassword()), true, userDTO.getRole());
+			User u = userRepository.save(user);
+			return Serializer.serializePlainJsonNode(u);
+		} else {
+			throw new BadRequestException(validationResult.getViolations().stream().collect(Collectors.joining(", ")));
+		}
+	}
+	
+	public JsonNode updateUser(String id, String loggedInUser, UserDTO userDTO) {
+		// check if updating password for own user
+		User user = userRepository.findById(id)
+				.orElseThrow(() -> new BadRequestException("User not found"));
+		
+		if(user.getEmail().equals(loggedInUser)) {
+				// current password matches with old provided
+//				user.setPassword(encoder.encode(userDTO.getNewPassword()));
+				user.setFirstName(userDTO.getFirstName() != null ? userDTO.getFirstName() : user.getFirstName());
+				user.setLastName(userDTO.getLastName() != null ? userDTO.getLastName() : user.getLastName());
+				user.setRole(userDTO.getRole() != null ? userDTO.getRole() : user.getRole());
+				userRepository.save(user);
+				return Serializer.serializePlainJsonNode(user);
+		} else {
+			log.error("Not allowed to change other user email");
+			throw new BadRequestException("Not allowed to change other user email");
+		}
+	}
+
+	public JsonNode updatePassword(String id, String loggedInUser, UserDTO userDTO) {
```

Comment:
Should we also add logic for admins to generate temporary passwords for users who forgot them?

Reply:
Perhaps even automated logic who sends an email with a temp password?

Reply:
This will require more complex solution than this, which include email server and other components.
Current solution is simple user & password management. We do not include even password history and similar restrictions (cannot use last x passwords)